lib/cli_util.dart

```diff
@@ -59,3 +59,66 @@ Directory getSdkDir([List<String>? cliArgs]) {
 
 /// Return the path to the current Dart SDK.
 String getSdkPath() => path.dirname(path.dirname(Platform.resolvedExecutable));
+
+/// Get the user-specific application configuration folder for the current
+/// platform.
+///
+/// This is a location appropriate for storing application specific
+/// configuration for the current user. The [productName] should be unique to
+/// avoid clashes with other applications on the same machine. This method won't
+/// actually create the folder, merely return the recommended location for
+/// storing user-specific application configuration.
+///
+/// The folder location depends on the platform:
+///  * `%APPDATA%\<productName>` on **Windows**,
+///  * `$HOME/Library/Application Support/<productName>` on **Mac OS**,
+///  * `$XDG_CONFIG_HOME/<productName>` on **Linux**
+///     (if `$XDG_CONFIG_HOME` is defined), and,
+///  * `$HOME/.config/<productName>` otherwise.
+///
+/// This aims follows best practices for each platform, honoring the
+/// [XDG Base Directory Specification][1] on Linux and [File System Basics][2]
```

Comment:
Do we have documentation for the windows convention?

Reply:
I tried, I can't find any canonical source on MSDN... The closes I got was:
https://docs.microsoft.com/en-us/windows/deployment/usmt/usmt-recognized-environment-variables

But I have no idea what that page actually pertains to.. It looks like it says:
>When using the XML files MigDocs.xml, MigApp.xml, and MigUser.xml, you can use environment variables to identify folders that may be different on different computers

So it pertains to people writing such XML files.. and not as general documentation of the Windows environment variables. Or at-least this is not clear to me :see_no_evil: 

There is also references to [Windows File System Namespace Usage Guidelines](https://www.scribd.com/document/48246732/Windows-File-System-Namespace-Usage-Guidelines) floating around, but again I can't find updated documentation from original source.

Who knows... maybe it's hidden behind an MSDN subscription? Or maybe it's just so widely known and there is so many search hits that finding the canonical documentation is very hard.

We could use wikipedia:
 * https://en.wikipedia.org/wiki/Environment_variable#Windows
 * https://en.wikipedia.org/wiki/Special_folder

I just gave up :rofl: